diag_manager/fms_diag_axis_object.F90

```diff
@@ -1174,6 +1196,68 @@ subroutine write_diurnal_metadata(this, fileobj)
                                       &trim(this%edges_name), str_len=len_trim(this%edges_name))
   end subroutine write_diurnal_metadata
 
+  !> @brief Creates a new z subaxis to use
+  subroutine create_new_z_subaxis(zbounds, var_axis_ids, diag_axis, naxis, file_axis_id, nfile_axis)
+    real(kind=r4_kind),                       intent(in)    :: zbounds(2)      !< Bounds of the Z axis
```

Comment:
Same real*4 comment.  The `zaxis_data` is a class(*) variable.  Shouldn't their types match?  Should these routines use the mixed-mode approach?

Reply:
The zbounds are read in from the yaml, I don't think you need to store them as r8s.
We just need to be sure to handle the conversion when calling `nearest_index`